kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,251 @@
+              addJobDoneListener {
+                dialog.dismiss()
+                if (status == Job.Status.SUCCEEDED) {
+                  showMapPreview(result)
+                } else {
+                  ("Job did not succeed: " + error.additionalMessage).also {
+                    Toast.makeText(this@MainActivity, it, Toast.LENGTH_LONG).show()
+                    Log.e(TAG, error.additionalMessage)
+                  }
+                }
+              }
+            }
+        } catch (e: InterruptedException) {
+          Toast.makeText(this, "TileCacheParameters interrupted: " + e.message, Toast.LENGTH_LONG).show()
+          Log.e(TAG, "TileCacheParameters interrupted: " + e.message)
+        } catch (e: ExecutionException) {
+          Toast.makeText(this, "Error generating parameters: " + e.message, Toast.LENGTH_LONG).show()
+          Log.e(TAG, "Error generating parameters: " + e.message)
+        }
+      }
+    }
+  }
+
+  /**
+   * Show tile cache preview window including containing the exported tiles.
+   *
+   * @param result takes the TileCache from the ExportTileCacheJob
+   */
+  private fun showMapPreview(result: TileCache) {
+    val newTiledLayer = ArcGISTiledLayer(result)
+    val previewMap = ArcGISMap(Basemap(newTiledLayer))
+
+    // set up the preview map view
+    previewMapView.apply {
+      map = previewMap
+      setViewpoint(mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE))
+    }
+    // control UI visibility
+    makeVisible(closeButton, dimBackground, preview_text_view, previewMapView)
+    exportTilesButton.visibility = View.GONE
+  }
+
+  /**
+   * Create a progress dialog box for tracking the export tile cache job.
+   *
+   * @param exportTileCacheJob the export tile cache job progress to be tracked
+   * @return an AlertDialog set with the dialog layout view
+   */
+  private fun createProgressDialog(exportTileCacheJob: ExportTileCacheJob): AlertDialog {
+    val builder = AlertDialog.Builder(this@MainActivity).apply {
+      setTitle("Exporting tiles...")
+      // provide a cancel button on the dialog
+      setNeutralButton("Cancel") { _, _ ->
+        exportTileCacheJob.cancel()
+      }
+      setView(R.layout.dialog_layout)
+    }
+    return builder.create()
+  }
+
+  /**
+   * Clear the preview window.
+   */
+  fun clearPreview(view: View) {
+    previewMapView.getChildAt(0).visibility = View.INVISIBLE
+    mapView.bringToFront()
+    // control UI visibility
+    makeVisible(exportTilesButton, mapView)
+  }
+
+  /**
+   * Makes the given views in the UI visible.
+   *
+   * @param views the views to be made visible
+   */
+  private fun makeVisible(vararg views: View) {
+    for (view in views) {
+      view.visibility = View.VISIBLE
+    }
+  }
+
+  /**
+   * Recursively deletes all files in the given directory.
+   *
+   * @param dir to delete
+   */
+  private fun deleteDirectory(dir: File?): Boolean {
```

Comment:
could just call File.deleteRecursively() here https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/delete-recursively.html

Reply:
nice one @sorenoid ! I found a .deleteRecursively attached directly to the cacheDir as well so have used that. Good spot cos it turned out I didn't have the creating of the cacheDir set up in the correct place anyway: I have fixed that now too.